datameta/api/metadata.py

```diff
@@ -158,9 +183,24 @@ def post(request:Request):
         isfile              = body["isFile"],
         submission_unique   = body["isSubmissionUnique"],
         site_unique         = body["isSiteUnique"],
+        service_id          = None if not service else service.id
     )
 
-    db = request.dbsession
     db.add(metadatum)
-
-    return HTTPNoContent()
+    db.flush()
```

Comment:
Same as above
```suggestion
```

Reply:
Don't I need the db.flush() to create the uuid/site id in this case?

Reply:
👍  Sorry, that was one removal too much then